umkc-cs-451-2020-spring/semester-project-group-4
# Setup React App, Express, MySQL, Update ReadMe, Add Component Library

Before submitting this PR, please make sure:

## What is This


## Ticket Number Or Ticket Link
#9 #8 #15 #7 #13 #14 


## Ticket AC
Initialize the Repo with React, Jest, Express, MySQL, Form builder, Material UI


## How to Test

Prerequisites
  1. Install node.js globally -- https://nodejs.org/en/
       npm i npm -g
  2. Install SQL Server 2019, Developer Edition -- https://www.microsoft.com/en-us/sql-server/sql-server-downloads  
     Do the initial setup. There is no need to change or install any extras. Make sure you can login via Windows/Mac/Linux Authentication.   
     (If you get stuck here, go ahead and move, it's not critical for now)  

Running Environment for Development
  1. cd/commerce_project
  2. npm i
  3. npm start
      This will allow you develop at http://localhost:3000  

Starting the Server
  1. cd/client/server
  2. npm i
  3. npm i mysql -g
  4. run `node server.js`
      This has the server(and MySQL) is running at http://localhost:3306

Running Tests (Jest)
  1. npm test
